lib/vagrant/plugin/v2/command.rb

```diff
@@ -230,6 +230,10 @@ def with_target_vms(names=nil, options=nil)
           color_index = 0
 
           machines.each do |machine|
+            if machine.state && machine.state.id != :not_created
```

Comment:
I don't know about this part. This will update the global index for pretty much any Vagrant command that is run. For large environments this might be non-negligibly large?

Reply:
Hmm, it seems like it would be better to look at what state is currently in the global index and compare it to what machines we know about, then adjust with the delta there. Then we wouldn't be doing this operation for every machine, only ones missing from the index file.

Reply:
And also for 99% of the time when there isn't anything missing, this will just become a noop which should save us any compute time